crypto/rand/randfile.c

```diff
@@ -286,8 +286,22 @@ const char *RAND_file_name(char *buf, size_t size)
         if (OPENSSL_strlcpy(buf, s, size) >= size)
             return NULL;
     } else {
+#ifdef OPENSSL_SYS_WINDOWS
+        /*
+         * We use the same env variables as GetTempFile() - but that function
+         * uses TCHARs, but getenv() gives us chars so its easier to do it this
+         * way
+         */
+        if ((s = getenv("TMP")) == NULL
+            && (s = getenv("TEMP")) == NULL
+            && (s = getenv("USERPROFILE")) == NULL
+            && (s = getenv("SYSTEMROOT")) == NULL) {
+            s = getenv("HOME");
```

Comment:
Shouldn't one choose variable that is more likely to be defined as last resort? I mean HOME is uncommon variable, in sense that Windows never defines one, and it would be the user who would choose to do that... Well, one can argue that given previous choices it will never get to HOME. But since it's something that user sets him-/herself, maybe it should be favoured first instead of last?